dali/operators/reader/loader/video/frames_decoder_test.cc

```diff
@@ -78,31 +89,100 @@ TEST_F(FramesDecoderTest, VariableFrameRate) {
     std::vector<uint8_t> frame(file.FrameSize());
 
     // Read first frame
+    ASSERT_EQ(file.CurrentFrame(), 0);
     file.ReadNextFrame(frame.data());
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 0), file.FrameSize());
 
     // Seek to frame
     file.SeekFrame(25);
+    ASSERT_EQ(file.CurrentFrame(), 25);
     file.ReadNextFrame(frame.data());
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 25), file.FrameSize());
 
     // Seek back to frame
     file.SeekFrame(12);
+    ASSERT_EQ(file.CurrentFrame(), 12);
     file.ReadNextFrame(frame.data());
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 12), file.FrameSize());
 
     // Seek to last frame (flush frame)
     file.SeekFrame(59);
+    ASSERT_EQ(file.CurrentFrame(), 59);
     file.ReadNextFrame(frame.data());
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 59), file.FrameSize());
 
     // Wrap around to first frame
     ASSERT_FALSE(file.ReadNextFrame(frame.data()));
     file.Reset();
+    ASSERT_EQ(file.CurrentFrame(), 0);
     file.ReadNextFrame(frame.data());
     this->CompareFrames(frame.data(), this->GetVfrFrame(1, 0), file.FrameSize());
 }
 
+TEST_F(FramesDecoderTest, VariableFrameRateGpu) {
+    ASSERT_TRUE(cuInitChecked());
+    CUdevice device = { 0 };
+    CUDA_CALL(cuDeviceGet(&device, 0));
+    CUcontext context = { 0 };
+    CUDA_CALL(cuCtxCreate(&context, 0, device));
+
+    std::string path = testing::dali_extra_path() + "/db/video/vfr/test_2.mp4";
+
+    // Create file, build index
+    FramesDecoderGpu file(path);
+
+    ASSERT_EQ(file.Height(), 600);
+    ASSERT_EQ(file.Width(), 800);
+    ASSERT_EQ(file.Channels(), 3);
+    ASSERT_EQ(file.NumFrames(), 60);
+
+    DeviceBuffer<uint8_t> frame;
+    frame.resize(file.FrameSize());
+    std::vector<uint8_t> frame_cpu(file.FrameSize());
+
+    for (int i = 0; i < 60; ++i) {
+        ASSERT_EQ(file.CurrentFrame(), i);
+        file.ReadNextFrame(frame.data());
+        copyD2H(frame_cpu.data(), frame.data(), frame.size());
+        this->CompareFramesAvgError(frame_cpu.data(), this->GetVfrFrame(1, i), file.FrameSize());
+    }
+    ASSERT_EQ(file.CurrentFrame(), -1);
+    file.Reset();
+    ASSERT_EQ(file.CurrentFrame(), 0);
+
+    // Read first frame
+    file.ReadNextFrame(frame);
+    copyD2H(frame_cpu.data(), frame.data(), frame.size());
+    this->CompareFramesAvgError(frame_cpu.data(), this->GetVfrFrame(1, 0), file.FrameSize());
+
+    // Seek to frame
+    file.SeekFrame(25);
+    file.ReadNextFrame(frame);
+    copyD2H(frame_cpu.data(), frame.data(), frame.size());
+    this->CompareFramesAvgError(frame_cpu.data(), this->GetVfrFrame(1, 25), file.FrameSize());
+
+    // Seek back to frame
+    file.SeekFrame(12);
+    file.ReadNextFrame(frame);
+    copyD2H(frame_cpu.data(), frame.data(), frame.size());
+    this->CompareFramesAvgError(frame_cpu.data(), this->GetVfrFrame(1, 12), file.FrameSize());
```

Comment:
Maybe you can seek to randomly selected frames from 1-58 range?

Reply:
Done

Reply:
I was thinking to replace `file.SeekFrame(25);` and `file.SeekFrame(12);` with that but that is fine.